src/main/java/sirius/kernel/timer/EndOfDayTaskExecutor.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.kernel.timer;
+
+import sirius.kernel.async.Tasks;
+import sirius.kernel.commons.Strings;
+import sirius.kernel.commons.TimeProvider;
+import sirius.kernel.commons.Watch;
+import sirius.kernel.di.PartCollection;
+import sirius.kernel.di.std.ConfigValue;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.di.std.Parts;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.health.Exceptions;
+import sirius.kernel.health.Log;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.IntPredicate;
+
+/**
+ * Executes all {@link EndOfDayTask end of day tasks} in the given timeframe.
+ *
+ * @see EndOfDayTask for a detailed description of the theory of operations
+ */
+@Register(classes = {EndOfDayTaskExecutor.class, EveryDay.class})
+public class EndOfDayTaskExecutor implements EveryDay {
+
+    @ConfigValue("timer.daily.end-of-day")
+    private int startHour;
+
+    @ConfigValue("health.endOfDayTasksLimit")
+    private int endHour;
+
+    @Part
+    private Tasks tasks;
+
+    @Part
+    private TimeProvider timeProvider;
+
+    @Parts(EndOfDayTask.class)
+    private PartCollection<EndOfDayTask> endOfDayTasks;
+
+    private Map<EndOfDayTask, EndOfDayTaskInfo> taskInfos = new ConcurrentHashMap<>();
+
+    @Override
+    public String getConfigKeyName() {
+        return "end-of-day";
+    }
+
+    @Override
+    public void runTimer() throws Exception {
+        tasks.defaultExecutor().start(this::executeTasks);
+    }
+
+    private void executeTasks() {
+        // As we limit the execution to a certain time frame we shuffle the execution on each run
+        // so that no single task can steal all the processing time...
+        List<EndOfDayTask> tasksToExecute = new ArrayList<>(endOfDayTasks.getParts());
+        Collections.shuffle(tasksToExecute);
+
+        IntPredicate endCondition = determineEndCondition();
+        for (EndOfDayTask task : tasksToExecute) {
+            int currentHour = timeProvider.localTimeNow().getHour();
+            if (endCondition.test(currentHour)) {
+                Log.BACKGROUND.INFO("Aborting end of day tasks, as the time limit has been hit...");
+                return;
```

Comment:
break?

Reply:
no there is no point in conitnuing , we reached the time limit -> abort all further processing